### PR TITLE
Hipchat API v2 Update

### PR DESCRIPTION
**Pull request by pdxmaverick**
I have updated code to use Hipchat's new APIv2.

Key differences, changed method to notification, updated exit code to 204, shortened From to just severity.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -14,7 +14,7 @@ either:
 
 for the latest release, or
 
-``pip install -e git+https://github.com/carsongee/zenoss-hipchat#egg=zenoss-hipchat``
+``pip install -e git+https://github.com/pdxmaverick/zenoss-hipchat#egg=zenoss-hipchat``
```
Comment (carsongee) [inline README.rst]: This shouldn't change

File: zenoss_hipchat/__init__.py [hunk 2]
```diff
@@ -5,4 +5,4 @@
 """
 
 
-VERSION = '0.2.0'
+VERSION = '2.0.0'
```
Comment (carsongee) [inline zenoss_hipchat/__init__.py]: This should just be 0.3.0

File: zenoss_hipchat/hipchat.py [hunk 3]
```diff
@@ -88,27 +88,26 @@ def _should_notify(self):
...
-        return 0
+            return True
+        return False
 
     def send(self):
         """
         Send off the message to environment room with a nice template
         """
-
+        print(self.post_url)
```
Comment (carsongee) [inline zenoss_hipchat/hipchat.py]: This doesn't need to be here

File: zenoss_hipchat/hipchat.py [hunk 4]
```diff
@@ -88,27 +88,26 @@ def _should_notify(self):
...
         if not self.clear:
             message = self._event_message()
         else:
             message = self._clear_message()
 
-        from_name = "{0} ({1})".format(
+        from_name = "{1}".format(
             config.HIPCHAT_FROM, self.SEVERITY_MAP[self.severity][1]
```
Comment (carsongee) [inline zenoss_hipchat/hipchat.py]: you should also remove the config.HIPCHAT_FROM argument to this format string

File: zenoss_hipchat/config.py [hunk 5]
```diff
@@ -5,12 +5,12 @@
...
-HIPCHAT_API_V1_TOKEN = os.environ.get('HIPCHAT_TOKEN', None)
+HIPCHAT_API_V2_TOKEN = os.environ.get('HIPCHAT_TOKEN', None)
 HIPCHAT_ROOM_ID = os.environ.get('HIPCHAT_ROOM', None)
-HIPCHAT_API_V1_ENDPOINT = os.environ.get(
+HIPCHAT_API_V2_ENDPOINT = os.environ.get(
     'HIPCHAT_API_ENDPOINT',
     'api.hipchat.com'
 )
-HIPCHAT_FROM = os.environ.get('HIPCHAT_FROM', 'Zenoss')
+HIPCHAT_FROM = os.environ.get('HIPCHAT_FROM', '')
```
Comment (carsongee) [inline zenoss_hipchat/config.py]: Why are you changing the default here? Seems to me, that it should stay the same

File: zenoss_hipchat/hipchat.py (hunk 4 above)
Comment (carsongee) [inline zenoss_hipchat/hipchat.py]: Actually, why change this at all?  You can always configure the FROM address yourself if you don't like seeing it come from `Zenoss`?  You could also add some logic to say:
```python
if config.HIPCHAT_FROM == '':
    from_name = self.SEVERITY_MAP[self.severity][1]
else:
    from_name = "{0} ({1})".format(
        config.HIPCHAT_FROM, self.SEVERITY_MAP[self.severity][1]
    )
```

I'd be fine with that

**Comment by carsongee** [comment]
Thanks for the contribution, and getting this up to date. I just have a few small change requests to accept it.  At the time I originally wrote this, APIv2 didn't allow you to set the from field, so I did APIv1. From the docs, it looked like that changed.

**Comment by pdxmaverick** [comment]
Thanks for the suggestions, I like the severity in the from, it is very useful.  The actual name is redundant IMO.  It might be a change in Hipchat since v1 was introduced.  The keys have a name that preceded the From data.  I have 4 different zenoss 5 environments feeding one alert stream room.  Each of the keys are name to represent the environment.  If I don't do it this way, there is too much data and the severity gets clipped off.

File: zenoss_hipchat/hipchat.py [hunk 6]
```diff
@@ -88,27 +88,34 @@ def _should_notify(self):
...
+            from_name = "{0}".format(
+            self.SEVERITY_MAP[self.severity][1]
+        )
+        else:
+            from_name = "{0} ({1})".format(
+            config.HIPCHAT_FROM, self.SEVERITY_MAP[self.severity][1]
+        )
+        
+        from_name = "{1}".format(
             config.HIPCHAT_FROM, self.SEVERITY_MAP[self.severity][1]
```
Comment (carsongee) [inline zenoss_hipchat/hipchat.py]: I think you want to remove the two lines above or they will overwrite everything done in the if/else above

**Comment by carsongee** [comment]
Other than that last little bit, this looks great.  I'll merge once that is fixed up.

**Comment by pdxmaverick** [comment]
Missed a line from my notes while updating, I cleaned it up.

Cheers

**Comment by carsongee** [comment]
Awesome, thanks for the updates

**Pull request merged**